add administration/forms (#154)
glpi-project/doc: source/modules/administration/forms/forms.rst

Forms
=====

Since GLPI 11, forms are now native, so the formcreator plugin is no longer required. Forms can be migrated from formcreator to forms.

Migration formcreator to forms
------------------------------

> .. warning:: Form migration must be done from the GLPI 10 database.

Author: Yes you're right. I've modified the first  sentence by  

> Since GLPI 11, forms are native, so the formcreator plugin is no longer required. Forms must be migrated from formcreator to forms.

> .. warning:: Form migration must be done from the GLPI 10 database.

Reviewer: The confusion still exists.
We say the migration has to be done from the GLPI 10 database, and then say to do the migration after updating to GLPI 11. Sounds like the warning needs changed to mention the form creator plugin needs to remain installed and that the migration of data from the plugin to the native form feature needs triggered manually.

> .. warning:: Form migration must be done from the GLPI 10 database.

Reviewer: yes

Once you have completed the migration from GLPI 10 to GLPI 11, simply enter the command from your GLPI folder:

``php bin/console migration:formcreator_plugin_to_core``

Forms by default
----------------

There are two pre-created forms by default:

* Report an issue
* Request a service

.. image:: images/forms_view_self-service.png
   :alt: view home self-service profile
   :scale: 37%

These 2 forms are accessible to everyone. They can be deactivated/deleted/modified

List of forms
-------------

The list of forms is available from **Administration** > **Forms**.
They will also be visible from **Assistance** > **Service catalog**. This tab refers to the self-service profile home page (see :doc:`Service catalog</modules/assistance/service-catalog>`).

.. image:: images/forms_list.png
   :alt: List of forms
   :scale: 42%

Forms options
-------------

When creating a form (by **+ Add**), several options will be available to customize and facilitate the user experience as much as possible.

.. image:: images/add_form.png
   :alt: Add a new form
   :scale: 42%

Customize the formatting
~~~~~~~~~~~~~~~~~~~~~~~~

> Tools are available to customize the organization of your forms

Reviewer: ```suggestion
Tools are available to customize the organization of the forms

```

.. image:: images/custom_formatting.png
   :alt: Customize the formatting
   :scale: 100%

#. Add a new question
#. Add a comment
#. Add a new section
#. Add an horizontal layout

> You can also move your questions to reorganize your form as you wish

Reviewer: ```suggestion
You can move the questions to reorganize the form

```

.. image:: images/move_question.png
   :alt: Reorganize your forms
   :scale: 80%

> You also have tools to reorganize an entire section

Reviewer: ```suggestion
There are tools to reorganize an entire section

```

.. image:: images/section_option.png
   :alt: Option for sections
   :scale: 69%

Basic informations
~~~~~~~~~~~~~~~~~~

* Modify the form's title in the **Underlined form** file (this title will be visible to people with access to the form)
* Add a **description** to simplify the use of the form (this description will be visible to people with access to the form)

Add questions
~~~~~~~~~~~~~

Each question has its own **title** field (identified by new question) and will be **visible to the user**.

> Each question **can be made mandatory** so that the user has no other choice than to fill in this field to validate their form

Reviewer: ```suggestion
Each question **can be made mandatory** so that the user must fill in the field to validate the form

```

"has no other choice" sounds like a threat

There are actually 11 possible question types:

> * **Short answer**: Text field with limited character size. This field has no formatting (bold, italics, color, etc.).

Reviewer: ```suggestion
* **Short answer**: Text field with limited length and no formatting (bold, italics, color, etc.).

```

.. image:: images/short_answer.png
   :alt: Short answer option
   :scale: 80%

This field has additional options:

* *Text*: Allows you to enter text (alphanumeric)
* *Emails*: Allows you to enter email addresses only (xxxxxx@xxxxxx.xx)
* *Number*: Allows you to enter numeric characters only

> * **Long answer**: unlimited text field in characters. The formatting of this field can be customized by the user (bold, italic, color, etc.)

Reviewer: Isn't there a DB column limit of 65,535 characters for all combined answers in a JSON format?

.. image:: images/long_answer.png
   :alt: Long answer option
   :scale: 80%

> * **Date and time**: field allowing you to insert a date and/or time

Reviewer: ```suggestion
* **Date and time**: a date and/or time

```

.. image:: images/date_and_time.png
   :alt: Date and time option
   :scale: 80%

This field has additional options:

> * *Current date/time*: Allows you to specify the current date and/or time by default and not allow the user to change it (leave the box unchecked to allow the user to specify the date/time they want).

Reviewer: ```suggestion
* *Current date/time*: Use the current date and/or time and not allow the user to change it (leave the box unchecked to allow the user to specify the date/time they want).

```

* *Date*: Specify the date only (without the time).
* *Time*: Specify the time of day.

.. tip: You can add a date only, a time only, or a combination of both.

> * **Actors**: allows you to add one or more people/groups/suppliers referenced in GLPI

Reviewer: ```suggestion
* **Actors**: add one or more users/groups/suppliers referenced in GLPI

```

.. image:: images/actors.png
   :alt: Actors option
   :scale: 80%

This field has additional options:

    * *Requesters / Observers / Assignees*: each users / groups / suppliers that will be added will be pre-filled in the ticket according to the type of actor that will be chosen.

>       You can also preselect a user/group/provider that the user can modify or not

Reviewer: ```suggestion
      A user/group/provider can be preselected that the user can modify or not

```

>     * *Allow multiple actors*: allow multiple selection (allows groups, users and providers in the same field)

Reviewer: ```suggestion
    * *Allow multiple actors*: allow multiple selections (allows groups, users and suppliers in the same field)

```

     .. warning:: If you want to allow the user to select multiple actor types, you will need to create as many questions as desired actors. An actor question can only contain one actor type.

> * **Urgency**: Allows you to add an urgency (very low, low, medium, high, very high)

Reviewer: ```suggestion
* **Urgency**: add an urgency (very low, low, medium, high, very high)

```

.. image:: images/urgency.png
   :alt: Urgency option
   :scale: 80%

> * **Request type**: Allows you to select the type of request (incident or request)

Reviewer: ```suggestion
* **Request type**: select the type of request (incident or request)

```

.. image:: images/request_type.png
   :alt: Request type option
   :scale: 80%

> * **Document**: Allow you to add a document

Reviewer: ```suggestion
* **Document**: add a document

```

.. image:: images/document.png
   :alt: Document option
   :scale: 80%

> * **Radio**: Allows you to set up a response list. The user will only be able to select one response

Reviewer: ```suggestion
* **Radio**: set up a response list. The user will only be able to select one response.

```

> You can add an additional description to make it easier for the user to choose

Reviewer: ```suggestion
An additional description can be added to make it easier for the user to choose.

```

.. image:: images/radio.png
   :alt: Radio option
   :scale: 80%

> * **Checkbox**: Allows you to set up a response list. The user will be able to select multiple answers

Reviewer: ```suggestion
* **Checkbox**: set up a response list. The user will be able to select multiple answers.

```

> You can add an additional description to make it easier for the user to choose

Reviewer: ```suggestion
An additional description can be added to make it easier for the user to choose.

```

.. image:: images/checkbox.png
   :alt: Checkbox option
   :scale: 80%

* **Dropdown**: set up a drop-down list to select one or more answers

.. image:: images/dropdown.png
   :alt: Dropdown option
   :scale: 80%

This field has an additional option:

  * *Allow muliple options*

* **Item**: Allows you to select differents GLPI's objects

.. image:: images/item.png
   :alt: Item option
   :scale: 80%

This field has additional options

**Users devices**: allows the user to select their own assets

**GLPI Objects** available :

> .. collapse:: Assets

Reviewer: These lists are going to be annoying to keep updated and may not be entirely useful to completely list out in this documentation.

Author: Possibly, I'd been asked to be as exhaustive as possible for the doc, and I thought that was one of the things I should mention.

    * Service packs
    * Operating systems architectures
    * Editions
    * Kernels
    * Kernels versions
    * Update sources

.. collapse:: Networking

    * Networking interfaces
    * Networks
    * Network port types
    * VLANs
    * Line operators
    * Domain types
    * Domain relations
    * Records types
    * Fiber types

.. collapse:: Cable management

    * Cables types
    * Cable strands
    * Socket models

.. collapse:: Internet

    * IP networks
    * Internet domains
    * Wifi networks
    * Networks names

.. collapse:: Software

    * Software categories

.. collapse:: User

    * Users titles
    * User categories

.. collapse:: Authorizations assignment rules

    * LDAP criteria

.. collapse:: Fields unicity

    * Ignored values for the unicity

.. collapse:: External authentications

    * Fields storage of the login in the HTTP request

.. collapse:: Power management

    * Plugs

.. collapse:: Appliance

    * Appliance types
    * Appliance environments

.. collapse:: Camera

    * Resolutions
    * Image formats

.. collapse:: Others

    * USB vendors
    * PCI vendors
    * Webhook categories

Configure visibility
~~~~~~~~~~~~~~~~~~~~

> Visibility conditions can be specified. This allows you to condition the display of a question based on the answer to a previous question.

Reviewer: ```suggestion
Visibility conditions can be specified for a question based on the answer to a previous question.

```

> For example, I'm reporting an issue with a printer.
> Depending on my location (question type **Item** > **Dropdowns** > **Common - Locations**), GLPI will be conditional on displaying only a list of printers linked to that location.

Reviewer: ```suggestion
For example, a user reports an issue with a printer.
Depending on the location (question type **Item** > **Dropdowns** > **Common - Locations**), the form could be conditional on displaying only a list of printers linked to that location.

```

Public access
~~~~~~~~~~~~~

Public access allows all users with the form link to access it in order to submit their request.
If you check the ``allow unauthenticated users`` box, you can also allow users who do not have a GLPI account to use the form.

.. tip:: If the ``allow unauthenticated users`` box is checked, authenticated users will no longer see this form

Private access
~~~~~~~~~~~~~~

Private access allows you to filter the visibility of the form.
It can be conditioned to users and/or groups and/or profiles. Multiple selection is allowed.

If all users should have access to the form, you can use the all users option in the drop-down list.

.. tip:: When your selection is complete, you will be able to see the list of people who have access to the form

   .. image:: images/matching_criteria.png
      :alt: List of users allowed to see the form
      :scale: 85%

If you see this message, This means that unauthenticated users are not allowed to see certain GLPI objects contained in certain questions.
You must then check the question(s) in order to modify them and avoid an error when submitting the form by unauthenticated users.

.. image:: images/error_form.png

>    :alt: Error form

Author: I don't agree. The documentation is heavy enough to read as it is, so the screenshots make it easier to read (I think) and it's quicker to understand where you stand. As for the alternative text, I suppose it's basically there for SEO purposes, but perhaps Mariia will take up the subject...

>    :alt: Error form

Reviewer: If the text is important enough to include in the documentation, it is important enough to be included as text (maybe stylized as a warning) rather than an image of the text which adds no extra context.
Alt text is an accessibility feature too, not just a marketing/SEO scheme. A user with a screen reader will have this image described as "Error form" instead of being read the actual alert text in the image.

>    :alt: Error form

Author: I edited the alt text and added an additional sentence for conflict resolution

   :scale: 42%

Import / Export
---------------

Forms allows exporting and importing forms between instances of GLPI.

.. Warning::This feature is designed to let administrators develop forms on a testing environment and copy them on a production environment.

Export
~~~~~~

To export forms, you need to use the massive actions

.. image:: images/export_forms.png
   :alt: Export forms
   :scale: 55%

Import
~~~~~~

To import forms, you need to click on **Import Forms** at the top of the screen

.. image:: images/import_forms.png
   :alt: Import forms
   :scale: 53%

If your import is correct, you can click on import

.. image:: images/import_ok.png
   :alt: Import correct
   :scale: 54%

Import behavior
^^^^^^^^^^^^^^^^

Forms allows field reconciliation.
If some information is missing (entity, user, group, etc.) from the instance where the form is imported,
Foms will offer you to either select a different existing value or create a new value for eache field in conflict..

.. image:: images/preview_import.png
   :alt: Import correct
   :scale: 50%

1. displays the field found during import
2. the value with which you wish to replace it

.. image:: images/reconciliation.png

>    :alt: Import correct

Author: Modified by : Form reconciliation

Forms
=====

> Since GLPI 11, forms are native, so the formcreator plugin is no longer required. Forms can be migrated from formcreator to forms.

Author: Since GLPI 11, forms are native, so the formcreator plugin is no longer required. Forms must be migrated from formcreator to forms.
